src/frontend/komponenter/Fagsak/Opprett/validator.ts

```diff
@@ -0,0 +1,62 @@
+import { feil, ok, ValideringsStatus } from '../../../typer/felt';
+import { IFøldselsnummerListeFelt, IFøldselsnummerFelt } from './typer';
+
+const harFyltInnFødselsnummer = (felt: IFøldselsnummerFelt): IFøldselsnummerFelt => {
+    return /^\d{11}$/.test(felt.verdi.replace(' ', ''))
+        ? ok(felt)
+        : feil(felt, 'Fødselsnummer.UgyldigFormat');
+};
+
+const fødselsnummerPassererMod10ogMod11Sjekk = (felt: IFøldselsnummerFelt): IFøldselsnummerFelt => {
+    const vektSifreMod10 = [3, 7, 6, 1, 8, 9, 4, 5, 2, 1];
+    const vektSifreMod11 = [5, 4, 3, 2, 7, 6, 5, 4, 3, 2, 1];
+
+    let sumMod10 = 0;
+    for (let i = 0; i < 10; i++) {
+        sumMod10 += Number.parseInt(felt.verdi.charAt(i), 10) * vektSifreMod10[i];
+    }
+
+    let sumMod11 = 0;
+    for (let i = 0; i < 11; i++) {
+        sumMod11 += Number.parseInt(felt.verdi.charAt(i), 10) * vektSifreMod11[i];
+    }
+
+    return sumMod10 % 11 === 0 && sumMod11 % 11 === 0
+        ? ok(felt)
+        : feil(felt, 'Fødselsnummer.ModSjekkFeil');
+};
+
+export const fødselsnummerValidator = (
+    fødselsnummerFelt: IFøldselsnummerFelt
+): IFøldselsnummerFelt => {
+    const validated = harFyltInnFødselsnummer(fødselsnummerFelt);
+    if (validated.valideringsStatus !== ValideringsStatus.OK) {
+        return validated;
+    }
+
+    return fødselsnummerPassererMod10ogMod11Sjekk(fødselsnummerFelt);
+};
+
+export const fødselsnummerListeValidator = (
+    fødselsnummerListeFelt: IFøldselsnummerListeFelt
+): IFøldselsnummerListeFelt => {
+    if (fødselsnummerListeFelt.verdi.length === 0) {
+        return feil(fødselsnummerListeFelt, 'Fødselsnummer.TomListe');
+    }
+
+    let firstError = -1;
+    const validatedListeFelt = {
+        ...fødselsnummerListeFelt,
+        verdi: fødselsnummerListeFelt.verdi.map((felt, index) => {
+            const validatedFelt = felt.valideringsFunksjon(felt);
```

Comment:
I think it should be possible to validate each child's fnr seperatly. Can you try to create an array with IFelt for each child? This way we can render an error message for each fnr.

Reply:
Yes, actually the fødselsnummerListeFelt is defined as IFelt<IFelt<string>[]>, which can hold both a global error message and individual messages for each child's value

Reply:
Oh, my bad 👍